streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamBranch.java

```diff
@@ -41,14 +43,18 @@
     private class KStreamBranchProcessor extends AbstractProcessor<K, V> {
         @Override
         public void process(final K key, final V value) {
-            for (int i = 0; i < predicates.length; i++) {
-                if (predicates[i].test(key, value)) {
+            for (int i = 0; i < predicates.size(); i++) {
+                if (predicates.get(i).test(key, value)) {
                     // use forward with child here and then break the loop
                     // so that no record is going to be piped to multiple streams
-                    context().forward(key, value, To.child(childNodes[i]));
-                    break;
+                    context().forward(key, value, To.child(childNodes.get(i)));
+                    return;
                 }
             }
+            // using default child node if supplied
```

Comment:
Should we really do it this way, or add a predicate to the list that always returns `true` if a default branch is added?